rewrite-yaml/src/main/java/org/openrewrite/yaml/ChangePropertyValue.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.yaml;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.Cursor;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Option;
+import org.openrewrite.Recipe;
+import org.openrewrite.Validated;
+import org.openrewrite.internal.NameCaseConvention;
+import org.openrewrite.internal.StringUtils;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.yaml.tree.Yaml;
+
+import java.util.Iterator;
+import java.util.Objects;
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class ChangePropertyValue extends Recipe {
+    @Option(displayName = "Property key",
+            description = "The key to look for. Glob is supported.",
+            example = "management.metrics.binders.*.enabled")
+    String propertyKey;
+
+    @Option(displayName = "New value",
+            description = "The new value to be used for key specified by `propertyKey`.")
+    String newValue;
+
+    @Option(displayName = "Old value",
+            required = false,
+            description = "Only change the property value if it matches the configured `oldValue`.")
+    @Nullable
+    String oldValue;
+
+    @Option(displayName = "Regex",
+            description = "Default false. If enabled, `oldValue` will be interepreted as a Regular Expression, and capture group contents will be available in `newValue`",
```

Comment:
@nmck257 If you have some time at some point, I'm not following what this parameter does. Specifically, I don't understand what `capture group contents will be available in newValue` means.

Reply:
Sure, quick example:
Input text: "hi I'm Nick"
oldValue: "hi I'm (.+)"
newValue: "hello $1"
regex: true
Output text: "hello Nick"

Reply:
Thanks -- appreciate it!